batch/test/test_batch.py

```diff
@@ -1070,3 +1070,101 @@ def test_job_private_instance_cancel(client: BatchClient):
     b.cancel()
     status = j.wait()
     assert status['state'] == 'Cancelled', str((status, b.debug_info()))
+
+
+def test_update_batch_no_deps(client: BatchClient):
+    bb = client.create_batch()
+    bb.create_job(DOCKER_ROOT_IMAGE, ['false'])
+    bb.submit()
+    j = bb.create_job(DOCKER_ROOT_IMAGE, ['true'])
+    b = bb.submit()
+    status = j.wait()
+
+    assert status['state'] == 'Success', str((status, b.debug_info()))
+
+
+def test_empty_update(client: BatchClient):
+    bb = client.create_batch()
+    bb.create_job(DOCKER_ROOT_IMAGE, ['true'])
+    b = bb.submit()
+    b = bb.submit()
+    status = b.wait()
+
+    assert status['state'] == 'success', str((status, b.debug_info()))
+
+
+def test_update_batch_w_submitted_job_deps(client: BatchClient):
+    bb = client.create_batch()
+    j1 = bb.create_job(DOCKER_ROOT_IMAGE, ['true'])
+    b = bb.submit()
+    j1.wait()
+    j2 = bb.create_job(DOCKER_ROOT_IMAGE, ['true'], parents=[j1])
+    b = bb.submit()
+    status = j2.wait()
+
+    assert status['state'] == 'Success', str((status, b.debug_info()))
+
```

Comment:
I think we also need tests here that try to submit updates to a batch that has been cancelled / deleted, in which case I suspect we get a 400 from the front-end